test/gui/TestLogParser.py

```diff
@@ -32,29 +32,61 @@ def traverse_loop(data):
+                                                "Error Details": str(
+                                                    outlineStep['detail']
+                                                )
+                                                if ('detail' in outlineStep)
+                                                else "Error details not found",
+                                            }
+                                            failing_tests.append(failing_test)
+                                    continue
+
                                 # Append the information of failing tests into the list of failing tests
-                                # If the error detail is missing(occurs mainly in runtime error) then we display the entire error object.
-                                test = {
-                                    "Feature File": str(feature_file['name']),
-                                    "Feature": str(feature['name']),
-                                    "Scenario": str(scenario['name']),
-                                    "Test Step": str(test_step['name']),
-                                    "Error Details": str(error['detail'])
-                                    if ('detail' in error)
-                                    else "Error details not found",
-                                }
-
-                                failing_tests.append(test)
+                                # If the error detail is missing(occurs mainly in runtime error) then we display "Error details not found" message.
+                                if 'result' in error and error['result'] == 'ERROR':
+                                    failing_test = {
+                                        "Feature File": str(feature_file['name']),
+                                        "Feature": str(feature['name']),
+                                        "Scenario": str(scenario['name']),
+                                        "Test Step": str(test_step['name']),
+                                        "Error Details": str(error['detail'])
+                                        if ('detail' in error)
+                                        else "Error details not found",
+                                    }
+                                    failing_tests.append(failing_test)
 
     return failing_tests
 
 
 def filter_redundancy(raw_data):
     unique_scenarios = []
+    unique_scenario_outline_examples = []
     filtered_data = []
 
     for scenario in raw_data:
-        if scenario['Scenario'] not in unique_scenarios:
+        if (
+            'Example' in scenario
+            and scenario['Example'] not in unique_scenario_outline_examples
+        ):
+            unique_scenario_outline_examples.append(scenario['Example'])
+            filtered_data.append(scenario)
+        elif 'Example' not in scenario and scenario['Scenario'] not in unique_scenarios:
```

Comment:
can you add a line of comment about this?

Reply:
Of course